content/en/docs/tasks/traffic-management/istioctl-describe/index.md

```diff
@@ -0,0 +1,241 @@
+   Port: http 9080/HTTP
+DestinationRule: ratings for "ratings"
+   Matching subsets: v1
+      (Non-matching subsets v2,v2-mysql,v2-mysql-vm)
+   Traffic Policy TLS Mode: ISTIO_MUTUAL
+Pilot reports that pod enforces HTTP/mTLS and clients speak mTLS
+{{< /text >}}
+
+The destination rule now appears in the output.  This tells us that the `ratings` destination rule is present, and that it defines the subset `v1` which matches this pod.
+Clients talking to the ratings microservice will use mutual TLS.
+
+## Virtual Services
+
+Now I will follow the Bookinfo example to [Request Routing](/docs/tasks/traffic-management/request-routing/) and define some virtual services:
+
+{{< text bash >}}
+$ kubectl apply -f samples/bookinfo/networking/virtual-service-all-v1.yaml
+{{< /text >}}
+
+After applying this rule I "describe" the _reviews-v1_ pod:
+
+{{< text bash >}}
+$ export REVIEWS_V1_POD=$(kubectl get pod -l app=reviews,version=v1 -o jsonpath='{.items[0].metadata.name}')
+$ istioctl x describe pod $REVIEWS_V1_POD
+{{< /text >}}
+
+The output resembles what we saw before the virtual services were defined, but we now see that they are present:
+
+{{< text plain >}}
+VirtualService: reviews
+   1 HTTP route(s)
+{{< /text >}}
+
+After applying _virtual-service-all-v1.yaml_ the traffic all goes to version 1.  The "stars disappear".  If this was a real cluster, someone might notice the logs to v2/v3
+are no longer appearing.  Users might notice features and not working.  `describe` will not
+just report the virtual services that configure a pod.  If it seems that a virtual service
+configures a pod, but actually blocks it, the output will include a warning.
+
+{{< text bash >}}
+$ export REVIEWS_V2_POD=$(kubectl get pod -l app=reviews,version=v2 -o jsonpath='{.items[0].metadata.name}')
+$ istioctl x describe pod $REVIEWS_V2_POD
+{{< /text >}}
+
+The warning "No destinations match pod subsets" tells us the problem.
+No traffic will arrive due to the virtual service destinations.
+
+{{< text plain >}}
+VirtualService: reviews
+   WARNING: No destinations match pod subsets (checked 1 HTTP routes)
+      Route to non-matching subset v1 for (everything)
+{{< /text >}}
+
+Oh no!  I must revert!  I'll delete the bogus Istio configuration:
+
+{{< text bash >}}
+$ kubectl delete -f samples/bookinfo/networking/destination-rule-all-mtls.yaml
+{{< /text >}}
+
+If I refresh the browser at this point the stars do not appear.  Instead I see
+*Error fetching product details!* and *Error fetching product reviews!*  Instead of
+panic, I `describe`:
+
+{{< text bash >}}
+$ istioctl x describe pod $REVIEWS_V2_POD
+{{< /text >}}
+
+{{< text plain >}}
+...
+VirtualService: reviews
+   WARNING: No destinations match pod subsets (checked 1 HTTP routes)
+      Warning: Route to UNKNOWN subset v1.  No DestinationRule.
+{{< /text >}}
+
+At this point I look back and realize I deleted the destination rules, not the virtual service V1 I wished to.  To fix the problem:
```

Comment:
I found it confusing when just reading this without opening up these yaml files.  would be good to highlight the corresponding yaml for the readers to connect the dots. :)

Reply:
i think the prob is the client side configured destination rule to talk to server in mTLS but the server didn't declare mTLS strict??